### ISSUE TITLE
Making croniter consistent with buggy Vixie/ISC/cronie crontabs

### ISSUE THREAD

**Issue by dswhite42** [pallets-eco/croniter#41]
Hello. Thank you very much for this library. 

There is a 30+ year old bug in Vixie Cron (and ISC cron and cronie and probably most other implementations) in how asterisks are parsed for day-of-month and day-of-week entries.  Officially, the crontab(5) man page says:
>Note: The day of a command's execution can be specified by two fields — day of month, and day of week.  If both fields are restricted (i.e., aren't *), the command will be run when  either  field  matches  the current time.  For example,
>`30  4  1,15 * 5` would cause a command to be run at 4:30 am on the 1st and 15th of each month, plus every Friday.

So when this code is run:
```python
import croniter, datetime
entry = '0 16 1,2,3 * FRI'
iterobj = croniter.croniter(entry, datetime.datetime(2023, 5, 1))
for i in range(0, 10):
    print(iterobj.get_next(datetime.datetime).strftime("%Y-%m-%d %H:%M:%S (%A)"))
```

Croniter prints out the correct dates: 4:00pm on May 1, May 2, May 3, plus every Friday
```
2023-05-01 16:00:00 (Monday)
2023-05-02 16:00:00 (Tuesday)
2023-05-03 16:00:00 (Wednesday)
2023-05-05 16:00:00 (Friday)
2023-05-12 16:00:00 (Friday)
2023-05-19 16:00:00 (Friday)
2023-05-26 16:00:00 (Friday)
2023-06-01 16:00:00 (Thursday)
2023-06-02 16:00:00 (Friday)
2023-06-03 16:00:00 (Saturday)
```

The bug in cron, however, is that it fails to parse day-of-month and day-of-week fields correctly if they contain an asterisk plus other characters, such as `*/2`.  The code below **looks like** it ought to indicate "4pm every other day (i.e. odd-numbered days), plus every Friday":

```python
import croniter, datetime
entry = '0 16 */2 * FRI'
iterobj = croniter.croniter(entry, datetime.datetime(2023, 5, 1))
for i in range(0, 10):
    print(iterobj.get_next(datetime.datetime).strftime("%Y-%m-%d %H:%M:%S (%A)"))
```

And that's what croniter thinks, too:
```
2023-05-01 16:00:00 (Monday)
2023-05-03 16:00:00 (Wednesday)
2023-05-05 16:00:00 (Friday)
2023-05-07 16:00:00 (Sunday)
2023-05-09 16:00:00 (Tuesday)
2023-05-11 16:00:00 (Thursday)
2023-05-12 16:00:00 (Friday)
2023-05-13 16:00:00 (Saturday)
2023-05-15 16:00:00 (Monday)
2023-05-17 16:00:00 (Wednesday)
```

but due to the bug, the actual parsed schedule on UNIX/Linux/Macs will be "4pm every other day (i.e. odd-numbered days), but only if that day is a Friday".   So the job will actually only run on May 5 (an odd-numbered Friday), not May 20, then again on May 27 and June 3, not June 10, etc.

(more details on the asterisk bug in cron is here: https://crontab.guru/cron-bug.html , and the consensus seems to be that this bug should **not** be fixed due to its having been around for decades now.)

So given that this bug in most every cron exists and is not going away, could croniter be updated to recognize what cron will actually do with entries like "0 16 */2 * FRI" ?

Or alternatively, if you don't want to change the default processing rules croniter already uses, could a `implement_cron_bug` flag be passed into the `croniter` class that will produce the buggy (but real-world) output?

Thanks again for your work on the tool.

**Comment by kiorky** [comment]
I m not willing to introduce a bug which will cause a major change in how croniter works for years also.

Also, i do not have much time and croniter is more or a maintenance mode, please note that if you contribute a PR where the bug is "handled" but only when a flag activates it, i ll review and merge it without any problem.

**Comment by kiorky** [comment]
fixed in 1.4.0 !

**Comment by dswhite42** [comment]
> fixed in 1.4.0 !

Great, thank you!
